java/arcs/core/data/Ttl.kt

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2020 Google LLC.
+ *
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ *
+ * Code distributed by Google as part of this project is also subject to an additional IP rights
+ * grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+package arcs.core.data
+
+/**  A class containing retention policy information. */
+data class Ttl(
+    val count: Int,
+    val units: Units?
+) {
+    enum class Units {
+        Minute,
+        Hour,
+        Day
+    }
+
+    fun isInfinite(): Boolean = this == Ttl.Infinite
+
+    override fun equals(other: Any?): Boolean =
+        other is Ttl && this.toMinutes() == other.toMinutes()
+
+    override fun hashCode(): Int = this.toMinutes().hashCode()
+
+    fun toMinutes(): Int {
+        when (this.units) {
+            Units.Minute -> return this.count
+            Units.Hour -> return this.count * 60
+            Units.Day -> return this.count * 60 * 24
+            else -> { // Note the block
+                throw IllegalArgumentException("Unsupported TTL units: ${this.units}")
+            }
+        }
+    }
```

Comment:
In addition to switching the `else` to `null` and changing the exception to `UnsupportedOperationException`, this could also be a property:

```kotlin
val minutes: Int
    get() = when (this.units) {
        Units.Minute -> return this.count
        Units.Hour -> return this.count * 60
        Units.Day -> return this.count * 60 * 24
        null -> { 
            throw UnsupportedOperationException("Units not available")
        }
    }
```

Reply:
Hmm... since `count` is a `val`, this could just be a straight-up property without the `get()`.

Reply:
updated the `get()`, not sure what you meant by the second comment - are you suggesting i add `minutes` as a property calculated during `init`? or something else? (sorry, this is my first ever kotlin code, there's a lot to learn)))

Reply:
Yeah, I think it could be calculated at init time.

```kotlin
val minutes: Int = count * when (units) {
    Units.Minute -> 1
    Units.Hour -> 60
    Units.Day -> 60 * 24
    null -> { 
        throw UnsupportedOperationException("Units not available")
    }
}
```